src/instrumenter/annotations.ts

```diff
@@ -56,75 +54,134 @@ function offsetBy(a: OffsetRange, b: number | OffsetRange): OffsetRange {
 
 let numAnnotations = 0;
 
-export class Annotation {
+/**
+ * Base class containing metadata for parsed anntotations useful for
+ * pretty-printing error messages and error line information.
+ */
+export class AnnotationMD<T extends SAnnotation> {
```

Comment:
I would still suggest to pick a better name than have an `MD` suffix. Even if `AnnotationMetaData` would occupy more space, it would look a bit more self-explanatory.